### PR TITLE
Vault marker games

### LINKED ISSUE #60: Refactor Marker Games to not use the GameMgr
The game manager is deprecated and will not be implemented on next-gen servers.

### PR DESCRIPTION
**Pull request by Hoikas**
This is part of H-uru/Plasma#312 and closes #60. Marker games are no longer stored using the game manager. Rather, they use the vault blob field. Quest games use chronicles to remember their state. CGZMs now store their scores in the authsrv for leaderboard generation.

This depends on H-uru/Plasma#515.

### CODE REVIEW

File: Python/grtzMarkerScopeGUI.py [hunk 1]
```diff
@@ -90,433 +67,294 @@
...
 
+        # Tear down game brain
+        PtSendKIMessageInt(kMGStopCGZGame, -1)
 
-
-    def IPartialRefreshDisplay(self):
-        "Refreshes only the volitile time entries (i.e. a game being played or a completed game)"
-        
+        # Update UI
+        self._supressNextNotify = True
```
Comment (Deledrius) [inline Python/grtzMarkerScopeGUI.py]: `suppress` :wink:

File: Python/grtzMarkerScopeGUI.py [hunk 2]
```diff
@@ -90,433 +67,294 @@
...
-        self.updateLevel = kNoRefresh
-        self.scopeID = -1
-        self.debug = 0
-        PtDebugPrint("__grtzMarkerScopeGUI: Max version %d - minor version %d" % (MaxVersionNumber,MinorVersionNumber))
+        self.version = 3
+
+        self._lookingAtGUI = False
+        self._pendingScoreUpdate = None
+        self._wantToUpdateGUI = False
+        self._supressNextNotify = False
```
Comment (Deledrius) [inline Python/grtzMarkerScopeGUI.py]: `suppress`

File: Python/grtzMarkerScopeGUI.py [hunk 3]
```diff
@@ -90,433 +67,294 @@
...
-                    PtDebugPrint("DEBUG: grtzMarkerScopeGUI.OnGUINotify():\tSelector value changed, stopping current game!")
-                    PtSendKIMessage(kMGStopCGZGame,0)
-                    self.updateLevel = kOneTimeFullRefresh
-                
-                #Reset the score if we've seleced a game
-                prevRefresh = self.updateLevel
-                if gameNum > -1 and gameNum < len(grtzMarkerGames.mgs):
-                    grtzMarkerGames.UpdateScore(gameNum,PtGetDniTime(),0.0)
-                    self.updateLevel = kPartialRefresh
+                if self._supressNextNotify:
```
Comment (Deledrius) [inline Python/grtzMarkerScopeGUI.py]: `suppress`

File: Python/grtzMarkerScopeGUI.py [hunk 4]
```diff
@@ -90,433 +67,294 @@
...
-                    timeTB.setString(ostring % (tuptime[3],tuptime[4],tuptime[5]))
-                else:
-                    ostring = "%02d:%02d:%02d:%02d"
-                    timeTB.setString(ostring % (tuptime[2]-1,tuptime[3],tuptime[4],tuptime[5]))
-            else:
-                timeTB.setString("--:--:--")
+        gameSelector.setValue(-1)
 
+    def _UpdateGUI(self, mission=-1, quitting=False, score=None, star=None):
+        # If missing is -1, this is a total update
```
Comment (Deledrius) [inline Python/grtzMarkerScopeGUI.py]: `missing == mission`?

**Comment by Deledrius** [comment]
:+1: Aside from the minor typos noted, this looks good. :smile:

File: Python/ki/__init__.py [hunk 5]
```diff
@@ -6702,92 +6419,76 @@ def ProcessNotifyMarkerFolderExpanded(self, control, event):
...
-                        # If playing, then the marker list is not selectable.
-                        if self.markerGameManager.gameData.data["svrGameTemplateID"] == self.markerGameDisplay.gameData.data["svrGameTemplateID"]:
-                            markerlistSelectable = False
-                    if markerlistSelectable:
-                        markerSel = control.getSelection()
-                        self.markerGameDisplay.setSelectedMarker(markerSel)
-                        self.BigKICheckContentRefresh(self.BKCurrentContent)
+                mgr.LoadGame(self.BKCurrentContent)
+                if not mgr.playing:
+                    # NOTE: We must use selected_marker_index because marker IDs don't nesecarily
```
Comment (Deledrius) [inline Python/ki/__init__.py]: `necessarily` :wink:

**Comment by Hoikas** [comment]
Fixed typos

**Pull request merged**